My Mission Statement as a Programmer (#131)
Bootstragram/bootstragram-blog: _posts/2024-01-06-developer-mission-statement.md

---
layout: post
id: 4c439aca-9f8f-4e66-9589-293f0481ab1a
title: My Mission Statement as a Programmer
author: Mick F
excerpt: >-
  Climate Friendliness, Human Friendliness, Exigence, IndieWeb Friendliness, Big
  Corporation Cautiousness, Durability & Sustainability serves as a guiding
  compass in my journey as a programmer
category: Column
tags:
  - iOS
  - Swift
---

## tl;dr

> As a programmer, I create eco-friendly software with user-friendly design,

Reviewer: ```suggestion
As a programmer, I create eco-conscious software with user-friendly design,
```

embracing IndieWeb principles. I'm cautious with big corporations, upholding

> ethics and user rights. My goal is durable, sustainable code, prioritizing

Reviewer: ```suggestion
ethics and user rights. My goal is to produce durable and sustainable code, prioritizing
```

Reviewer: or "write" instead of "produce"

maintainability and developer well-being. I respond swiftly to evolving
challenges for impactful work in a changing tech landscape.

## A longer version

Let me explain. There is a sense of uneasiness with the state of the world these

> days. And after I had the privilege to spend a fun and peaceful holiday season,

Reviewer: ```suggestion
days. After having the privilege to spend a fun and peaceful holiday season,
```

> I also wanted to give a headcheck on where I stand with my role as a programmer

Reviewer: ```suggestion
I wanted to do a headcheck on where I stand with my role as a programmer
```

Reviewer: Not sure if "headcheck" is the right word, but we get it and I can't think of a better term

and if I am aligned with my aspirations and values. So I decided to write my

> mission statement. Let me develop each of its components, with an example of how

Reviewer: ```suggestion
mission statement. I will develop each of its components with an example of how
```

I think I apply it.

### 🌍 Climate Friendliness

I need to consider all environmental impacts of the code I write: the hardware
it runs on, the energy it requires, its purpose for the people running it.

> Example. I work for [Memo Bank][18], a bank for companies, that [acts][1]

Reviewer: ```suggestion
Example. I work for [Memo Bank][18], a bank for small and medium-sized companies, that [prioritizes in a no bull-shit way][1]
```

> towards the transition to a low carbon economy. It has been recognised as such

Reviewer: ```suggestion
the transition to a low carbon economy. It has been recognised as such
```

Reviewer: ```suggestion
towards the transition to a low carbon economy. It has been recognised as for its approach
```

Reviewer: ```suggestion
towards the transition to a low carbon economy. It has been recognised for its progressive approach
```

Reviewer: Maybe add a little conclusion on this point? something like : "This is one of the main values I really appreciate about working at Memo Bank. Not only does it make me proud to work here, but it helps me learn and stay informed about this important issue, as Memo is continuously providing training and other resources on the subject for us employees.

> by [Reclaim Finance][2], and is recommended on their website [Change de

Reviewer: ```suggestion
by [Reclaim Finance][2], and is one of the recommended banks on their website [Change de
```

banque][3].

### 🤝 Human Friendliness

I prioritise security and user experience, in that order, to promote the
well-being of end-users. It involves creating software that is intuitive,
inclusive, and positively impacts users' lives.

Example. I feel that it was my main motivations to start running the [Statium

> Newsletter][4], a minimalistic soccer newsletter written in French that focus on

Reviewer: ```suggestion
Newsletter][4], a minimalistic soccer newsletter written in French that focuses solely on
```

the game. It filters out the noise from meaningless controversies, [poor
journalistic practices][5], invasive advertisement and poor user experience of
alternative sources. As of today, [statium.app][6] has a perfect score of 100 in
Performance, Accessibility, Best Practices and SEO with Lighthouse.

> ### ⚡ Exigence

Reviewer: Not sure we use this word in English though it's so good. Maybe "Rigor"?

> I want to continuously get better at my craft. There is no done software. A

Reviewer: ```suggestion
I want to continuously get better at my craft. Software is never "done". A
```

skillset is never complete. I want to keep engaging on how to be a better
programmer, a better team member, and a better person.

Example. I think a good practice of exigence is balancing being very curious
with just what it needs of focus. I wrote about [why side-projects are so
important to me][7] in the past: my curiosity is satisfied and the new things I
learn have many positive impacts on my job at Memo Bank.

### 🌐 IndieWeb Friendliness

I support decentralised and people-focused technologies and products that help
individuals to have greater control over their online presence and their data.

Example. I use RSS daily, I [POSSE][8], I run a newsletter using
[Buttondown][9], my microblogging platform of choice is [micro.blog][10], I
favor community-run or independant sources when I use links in my blog posts
(examples: [TMDB][11], [Discogs][12], [Sports Reference][13], etc.). I make [my
default apps][14] public.

### 🕵️‍♂️ Big Corporation Cautiousness

I am curious of the ethical considerations of big companies when engaging with

> their products and technologies. I avoid those that compromise over user privacy

Reviewer: ```suggestion
their products and technologies. I avoid those that compromise user privacy
```

or promote monopolistic behavior.

Example. I stopped using Gmail, I don't actively use Meta products or X. I am

> wary of AI products and try to use them sparingly and as ethically as possible.

Reviewer: ```suggestion
wary of AI products and try to use them as sparingly and ethically as possible.
```
